tests/test_uint256.cpp

```diff
@@ -0,0 +1,779 @@
+/**
+* @brief
+* @file test_uint256.cpp
+* @author C.M.
+* @date 2018-10-21
+*/
+
+/* module header */
+
+/* system includes C */
+
+/* system includes C++ */
+#include <gtest/gtest.h>
+
+
+/* local includes */
+#include "helpers/uint256.h"
+
+// TODO: Check if the logic implemented in writeu128BE is correct
+TEST(TEST_UINT256, WRITEU128BE)
+{
+	uint128_t number;
+	uint8_t buffer;
+
+	UPPER(number) = 0x0;
+	LOWER(number) = 0x123456789abcde;
+	writeu128BE(&number, &buffer);
```

Comment:
the buffer here is too short (should be at least `sizeof(uint128_t)`), you are writing beyond its boundary into an invalid memory space.

Reply:
Calling  _writeu128BE(&number, &buffer)_ with this values:
    ```number (LOWER 123456 UPPER 89abcd) BUFFER: 0 ```
return
    ```number (LOWER: 123456 UPPER: 0) BUFFER: 0 ```
Is the correct reply for writeu128be?
Or should be one of this:
```  number (LOWER cdab89 UPPER 563412) BUFFER: 0 ``` 
```  number (LOWER 563412 UPPER cdab89) BUFFER: 0  ```

Reply:
BTW, uint128_t is composed by two uint64_t which is unsigned long int with limit of 4,294,967,295 decimal or in hexa FFFFFFFFFFFFFFFF, we can check it running this test
```
	UPPER(n1) = 0x1;
	LOWER(n1) = 0;
	UPPER(n2) = 0;
	LOWER(n2) = 0xFFFFFFFFFFFFFFFF;
	minus128(&n1, &n2, &result);
	ASSERT_EQ(UPPER(result), 0);
	ASSERT_EQ(LOWER(result), 1);
```
So LOWER(number) = 0x123456789abcde; its into their boundary. Right?

Reply:
this still needs fixing: `writeu128BE` encodes a big endian uint128 into a buffer. The buffer length must be defined as `uint8_t buffer[sizeof(uint128_t)]`